forwarder.go

```diff
@@ -0,0 +1,70 @@
+package k8sportforward
+
+import (
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type ForwarderConfig struct {
+	RestConfig *rest.Config
+}
+
+type Forwarder struct {
+	restConfig *rest.Config
+
+	k8sClient rest.Interface
+}
+
+func NewForwarder(config ForwarderConfig) (*Forwarder, error) {
+	if config.RestConfig == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.RestConfig must not be empty", config)
+	}
+
+	setConfigDefaults(config.RestConfig)
+
+	k8sClient, err := rest.RESTClientFor(config.RestConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	f := &Forwarder{
+		restConfig: config.RestConfig,
+
+		k8sClient: k8sClient,
+	}
+
+	return f, nil
+}
+
+// ForwardPort opens a tunnel to a kubernetes pod.
+func (f *Forwarder) ForwardPort(namespace string, podName string, remotePort int) (*Tunnel, error) {
```

Comment:
The interface changed. Before we accepted the tunnel config, which was weird. The tunnel is something we compute internally. The tunnel config is now cleaner.